assets/src/edit-story/components/canvas/singleSelectionMovable.js

```diff
@@ -190,9 +190,11 @@ function SingleSelectionMovable({ selectedElement, targetEl, pushEvent }) {
           };
           updateSelectedElements({ properties });
           if (selectedElement.resource) {
-            handleDrop(selectedElement.resource, selectedElement.id);
+            await handleDrop(selectedElement.resource, selectedElement.id);
           }
         }
+        setIsDragging(false);
```

Comment:
Hmm. So, these are happening outside the event phase? Each of the ones below will force rerender thrashing. I recommend to create a simple reset callback using the `useBatchingCallback` and call it from `handleDrop().then()` to make it obvious.

Reply:
Done.